feat: add visor docs (#6996)
vegaprotocol/vega: visor/readme.md

# Vega Visor

A binaries runner for [Core](../core/README.md) and [Data Node](../datanode/README.md) that facilitates automatic protocol upgrades.

***Features:***
- Visor is connected to core nodes and listens for protocol upgrades.
- Visor allows you to run core and data node binaries based on custom configuration.
- When a protocol upgrade is ready to be used, Visor automatically stops currently running binaries and starts new ones with the selected version.
- Visor can be configured to automatically fetch binaries with the correct version during the upgrade.
- Visor is highly configurable and allows you to configure the number of restarts, restarts delays, specific upgrade configuration and much more.

## Architecture

> Visor stores all it's required config files and state in a `home` folder. The basic folder structure can be geneate by `visor init` cmd or manually.

Reviewer: ```suggestion
Visor stores all its required config files and state in a `home` folder. The basic folder structure can be generated by `visor init` cmd or manually.
```

3. Visor connects to the core node and communicates with it via RPC API.
4. When validators agree on executing a protocol upgrade to a certain `version` and the network has reached the proposed `upgrade block height`, the core node will notify Visor about the upgrade.
- When `autoInstall` is enabled by a validator, then Visor automatically fetches the binaries with the correct `version` and prepares the upgrade folder.
- When `autoInstall` is NOT enabled, then a validator has to manually download the right binaries with correspoding `version` and prepare the upgrade folder with `run-config.toml` in it before the `upgrade block height`.
5. When Visor is notified about the upgrade to a specific `version`, it links the upgrade folder assosiated with the upgrade (either by being called as `version` or being specifically mapped manually in [Visor config](#visorconfiguration)). Then it executes the run config from the upgrade folder.

After that, the whole process is repeated from points 3-5 every time another upgrade takes place.

## Setup

> Visor stores all it's required config files and state in a `home` folder. The basic folder structure can be geneate by `visor init` cmd or manually.

Reviewer: ```suggestion
Visor stores all its required config files and state in a `home` folder. The basic folder structure can be generated by `visor init` cmd or manually.
```

> It is vital that all necessery files and folders are present in the `home` folder, therefore using the `init` command is recommended.

Reviewer: ```suggestion
It is vital that all necessary files and folders are present in the `home` folder, therefore using the `init` command is recommended.
```

> It is recomened to run Visor itself with a service manager for example Systemd.

Reviewer: ```suggestion
It is recommended that you run Visor itself with a service manager such as Systemd.
```
